app/api/items/add_item.feature

```diff
@@ -2,74 +2,209 @@ Feature: Add item
+    |                  ID | idItemParent | idItemChild         | iChildOrder |
+    | 4037200794235010051 |           21 | 5577006791947779410 |         100 |
+  And the table "items_ancestors" should be:
+    | idItemAncestor | idItemChild         |
+    | 21             | 5577006791947779410 |
+  And the table "groups_items" at ID "8674665223082153551" should be:
+    |                  ID | idGroup | idItem              | idUserCreated | ABS(sFullAccessDate - NOW()) < 3 |  bOwnerAccess | bManagerAccess | ABS(sCachedFullAccessDate - NOW()) < 3 | bCachedFullAccess |
+    | 8674665223082153551 |      11 | 5577006791947779410 |             1 | 1                                |             1 |              1 | 1                                      |                 1 |
 
-Scenario: Id not given
-  Given I am the user with ID "1"
-  When I send a POST request to "/items/" with the following body:
+
+  Scenario: Valid (all the fields are set)
+    Given I am the user with ID "1"
+    And the database has the following table 'groups':
+      | ID    |
+      | 12345 |
+    And the database has the following table 'groups_ancestors':
+      | ID | idGroupAncestor | idGroupChild | bIsSelf |
+      | 73 | 12              | 12345        | 0       |
+    And the database has the following table 'items':
+      | ID |
+      | 12 |
+      | 34 |
+    And the database has the following table 'groups_items':
+      | ID | idGroup | idItem | bManagerAccess | bOwnerAccess |
+      | 42 | 11      | 12     | true           | false        |
+      | 43 | 11      | 34     | false          | true         |
+    And the time now is "2018-01-01T00:00:00Z"
+    When I send a POST request to "/items/" with the following body:
+      """
+      {
+        "item": {
+          "type": "Course",
+          "url": "http://myurl.com/",
+          "text_id": "Task number 1",
+          "title_bar_visible": true,
+          "custom_chapter": true,
+          "display_details_in_parent": true,
+          "uses_api": true,
+          "read_only": true,
+          "full_screen": "forceYes",
+          "show_difficulty": true,
+          "show_source": true,
+          "hints_allowed": true,
+          "fixed_ranks": true,
+          "validation_type": "AllButOne",
+          "validation_min": 1234,
+          "unlocked_item_ids": "12,34",
+          "score_min_unlock": 34,
+          "team_mode": "All",
+          "teams_editable": true,
+          "team_in_group_id": "12345",
+          "team_max_members": 2345,
+          "has_attempts": true,
+          "access_open_date": "2018-01-02T03:04:05Z",
+          "duration": "01:02:03",
+          "end_contest_date": "2019-02-03T04:05:06Z",
+          "show_user_infos": true,
+          "contest_phase": "Analysis",
+          "level": 345,
+          "no_score": true,
+          "group_code_enter": true
+        },
+        "language_id": "3",
```

Comment:
this format is bit strange with "item", "language_id", "string" at the root. It looks like something constraints by technical reasons ;-)

Reply:
fixed